### DISCUSSION TITLE
Autologin on LTSP 23.01 on Debian Bullseye with MATE DE

### DISCUSSION THREAD

**Discussion by N0rbert** [ltsp/ltsp#893] [Q&A]
I'm using LTSP 5 on 150+ client machines for about 12 years with MATE DE starting from Ubuntu 12.04 LTS and then 16.04 LTS and Debian 9. 
So first of all I would like to thank the community and personally @alkisg for this great and useful project!

Now I'm migrating my labs to modern LTSP with Debian 11.

During my tests I have discovered an issue with LigthDM auto-login.

## Installation and configuration

I have installed from *debian-11.6.0-amd64-netinst.iso* for LTSP server inside VirtualBox with nested VT-x and 128 Gb hard disk.
During the installation on *Software selection* step I have checked *MATE* and *standard system utilities*.

Then I did basic configuration of the installed system as *root* user:

```
sed -i 's/main/main contrib non-free/g' /etc/apt/sources.list
apt-get update
apt-get install ssh mc htop imagemagick
```

then I have logged as normal user and enabled `sudo` by

```
su -l -c "usermod -a -G sudo $USER && reboot"
```

then I have added LTSP repository, installed LTSP and configured it:

```
sudo apt-get install software-properties-common
sudo apt-key adv --recv-keys --keyserver keyserver.ubuntu.com 4F3A3E9FC07F4E831D87AC2D3C72607B03AFA832
sudo add-apt-repository "deb http://ppa.launchpad.net/ts.sch.gr/debian/ubuntu focal main"
sudo apt-get update

sudo apt-get install --install-recommends ltsp ltsp-binaries dnsmasq nfs-kernel-server openssh-server squashfs-tools ethtool net-tools epoptes
sudo gpasswd -a $USER epoptes

sudo ltsp dnsmasq
sudo ltsp nfs
sudo ltsp ipxe
```

Booted real laptop using PXE to ensure that Memtest is working.

Then I made an image from already installed server system:

```
sudo install -m 0660 -g sudo /usr/share/ltsp/common/ltsp/ltsp.conf /etc/ltsp/ltsp.conf

sudo ltsp image /
sudo ltsp ipxe
sudo ltsp initrd
```

and checked that *x86_64.img* is booting normally to LightDM login screen from PXE menu.

## Manual login

The next step was a creation of new user named *tester* with default set of groups:

```
sudo adduser tester # use "tester" without qoutes as password
sudo usermod -a -G cdrom,floppy,audio,dip,video,plugdev,netdev,lpadmin,scanner tester
sudo ltsp initrd
```

After this I have booted test PXE machine and was able to login to MATE session using *tester* account. 
So normal manual login is working.

## Autologin

Then I tried to configure auto-login as follows:

```
cat <<EOF | sudo tee /etc/ltsp/ltsp.conf
# /bin/sh -n
[server]

[common]

[clients]

[54:42:49:xx:xx:xx]
HOSTNAME=test
AUTOLOGIN=tester
PASSWORDS_TEST="tester/dGVzdGVyCg=="
EOF

sudo ltsp initrd
```

Then I have booted the PXE client.
On the LightDM I see user selection drop-down menu with *tester* in it. 

![loginscreen](https://github.com/ltsp/ltsp/assets/309456/d26ab20f-02d8-439e-b5cb-2281e612bd36)

So I have selected *tester* and hit *Log In*. **Please note this moment - on Debian Bullseye I got interactive "autologin"**.

Previously I have already discovered that there are some differences between LightDM in Debian and in Ubuntu. For example LightDM in Debian does not source `~/.profile` file (see https://github.com/ubuntu/ubuntu-make/issues/678 and [linked Debian bug](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1024073)). Maybe current issue have the same reasons underneath.

Have anyone already tested autologin into LightDM on Debian (11/bullseye in my case) with MATE DE?

Archives with logs and for server and client are attached.
[server.tar.gz](https://github.com/ltsp/ltsp/files/14348150/server.tar.gz)
[client.tar.gz](https://github.com/ltsp/ltsp/files/14348151/client.tar.gz)
LTSP packages and epoptes versions are 23.01-1~ubuntu20.04.1 .

Update: autologin using *slick-greeter* does not work either (as it is based on *lightdm*), autologin using *gdm3* works but I do not like it.

---

My current workaround is the following: 

1. install full new MATE system to raw disk file at `/srv/ltsp/debian.img` with the same above package selection;
1. install `sddm` package there;
1. remove the problematic *lightdm-xsession* desktop session file by running `rm -v /usr/share/xsessions/lightdm-xsession.desktop`;
1. build the squashfs image using `sudo bash -c "ltsp image debian && ltsp kernel debian && ltsp ipxe && ltsp initrd"`;
1. finally boot the client to get normal auto-login to `DESKTOP_SESSION=mate` (to allow Caja to manage desktop).

**Comment by vagrantc** [comment]
Curious why you are using the old Debian bullseye (11), as Debian bookworm (12) was released middle of last year already... is there a specific reason the newer version won't work for you?

I am not sure there is anything in the PPA that is not already present in Debian bookworm's native ltsp packages. Possibly a difference with ltsp-binaries...

**Reply by N0rbert** [reply]
Thanks for reply! 
But I need this specific Debian supported *oldstable* version because of several reasons. 
AFAIS the differrencies between LTSP 23.02 and 23.01 are minimal.
